packages/qdrant-loader/src/qdrant_loader/config/__init__.py

```
@@ -399,6 +399,35 @@ def _substitute_env_vars(data: Any) -> Any:
             return [Settings._substitute_env_vars(item) for item in data]
         return data
 
+    @staticmethod
+    def _validate_env_substitution(data: Any) -> None:
```

Ruff (0.15.14)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `data`

(ANN401)